backend/src/apiserver/storage/pipeline_store.go

```diff
@@ -175,38 +221,103 @@ func (s *PipelineStore) DeletePipeline(id string) error {
 }
 
 func (s *PipelineStore) CreatePipeline(p *model.Pipeline) (*model.Pipeline, error) {
+	// Set up creation time for both pipeline and pipeline version.
 	newPipeline := *p
 	now := s.time.Now().Unix()
 	newPipeline.CreatedAtInSec = now
+	newPipeline.DefaultVersion.CreatedAtInSec = now
```

Comment:
the defaultversion is not guarantee to exist. this might cause nil pointer exception.

Reply:
Here is the tricky part: 
(1) in the final release, CreatePipeline in pipline_store shouldn't refer or populate the version part at all, since it should only the Pipeline container.
(2) in the intermediate status, CreatePipeline has to create an implicit version with pipeline, so what did is when resource_manager calls CreatePipeline in pipeline_store, I manually put a version in the Pipeline struct and hence here the default version always exist. BUT the problem is that CreatePipeline defined in pipeline_store is called not via resource_manager, e.g., directly called from cli? I am not sure if there is such a case. Do you have any idea. If not, then we are fine and can be sure DefaultVersion is not nil here.